specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+
+**GetSharedWorkers**: The GetSharedWorkers method, part of the CoreWebViewSharedWorkerManager class, is used to retrieve all shared workers created within the context of a WebView2 application for the profile. This method gives a collection of CoreWebView2SharedWorkers objects.
+
+# Examples
+## Dedicated Worker
+## Monitoring the Creation/Destruction of Dedicated Workers.
+The following example demonstrates how to subscribe to the event that is triggered when a web page creates a dedicated worker. This event provides a `CoreWebView2DedicatedWorker` object, enabling the host application to interact with it.
+
+This example also showcases the utilization of the upcoming PostMessage and WorkerMessageReceived APIs. These APIs enable efficient communication between the main thread and a dedicated worker.
+
+## .NET/WinRT
+```c#
+    void PostMessageToWorker(CoreWebView2DedicatedWorker worker)
+    {
+        // Inside the worker, an event listener is set up for the 'message' event. When a message is received from the host application, the worker performs a fetch request. The host application also sets up an event listener to receive messages from the worker. These messages could contain fetched data or any error information from the worker's fetch operation.
+        worker.WorkerMessageReceived += (sender, args) =>
+        {
+            var message = args.TryGetWebMessageAsString;
+            // This section of the code is dedicated to handling updates received from the worker. Depending on the nature of these updates, you might choose to perform various actions. For instance, you could modify the user interface to reflect the new data, or you could log the received data for debugging or record-keeping purposes.
+        };
+
+        // You can dispatch a message to the worker with the URL you want to fetch data from the host app.
+        worker.PostMessage("{type: 'FETCH_URL',url: 'https://example.com/data.json'");
+    }
+
+    void DedicatedWorkerCreatedExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        webView.CoreWebView2.DedicatedWorkerCreated += (sender, args) =>
+        {
+            CoreWebView2DedicatedWorker dedicatedWorker = args.Worker;
+            if(dedicatedWorker != null)
+            {
+              dedicatedWorker.WorkerDestroyed += (sender, args) =>
+              {
+                 /*Cleanup on worker destruction*/
+              };
+
+                var scriptUri = dedicatedWorker.ScriptUri;
+                  MessageBox.Show("Dedicated is created at" + scriptUri , "Dedicated Worker Message");
```

Comment:
Probably doesn't make sense to pop a dialog for this in an actual application. Just the PostMessage should be good. Removing MessageBox here also means you don't have to make the MessageBox.Show async to avoid running message loop in a WebView2 event handler.

Also applies below to other samples using message box